### PR TITLE
perf(ext/http): Use flat list of headers for multiple set/get methods

### PR DESCRIPTION
**Pull request by kamilogorek**
This PR attempts to resolve the first item on the list from https://github.com/denoland/deno/issues/19330 which is about using a flat list of interleaved key/value pairs, instead of a nested array of tuples.

I can tackle some more if you can provide a quick example of using raw v8 arrays, cc @mmastrac

### CODE REVIEW

File: ext/http/http_next.rs [hunk 1]
```diff
@@ -245,9 +245,7 @@ pub fn op_http_get_request_header(
 }
 
 #[op]
-pub fn op_http_get_request_headers(
-  slab_id: SlabId,
-) -> Vec<(ByteString, ByteString)> {
+pub fn op_http_get_request_headers(slab_id: SlabId) -> Vec<ByteString> {
   let http = slab_get(slab_id);
   let headers = &http.request_parts().headers;
   let mut vec = Vec::with_capacity(headers.len());
```
Comment (bartlomieju) [inline ext/http/http_next.rs:251]: This should be doubled, or maybe even slightly overallocated to avoid reallocs when there's "cookies" header

Comment (kamilogorek) [reply ext/http/http_next.rs]: Updated. No need to preallocate for cookies, as this header is created as a part of headers map iteration, which mean that if there is one, it'll skipped the initial header `push` and will be added later. This mean that there can be max `headers.len() * 2` entries, even with cookies.

**Comment by mmastrac** [comment]
> I can tackle some more if you can provide a quick example of using raw v8 arrays, cc @mmastrac

@kamilogorek Awesome, thanks for the contribution. Let's land this and we can do the v8::Array work in a follow-up PR.

Please take a look at how we create the raw v8 arrays in `runtime.rs` (https://github.com/denoland/deno/blob/main/core/runtime.rs#L2508). We start off with a `SmallVec` and fill it with values. In this case, these are Latin-1 headers which means you can use V8's 1-byte strings (https://docs.rs/rusty_v8/latest/rusty_v8/struct.String.html#method.new_from_one_byte). Finally, you can use `new_with_elements` to make the array, and return that from the op (https://docs.rs/rusty_v8/latest/rusty_v8/struct.Array.html#method.new_with_elements).

**Comment by bartlomieju** [comment]
I would suggest we first bench it and see what are the gains before working on migrating other places in code. @mmastrac could you take a look if there are any RPS/latency improvements with this patch?

**Comment by mmastrac** [comment]
Perf test (M2 mac, wrk run):

Before: 117.5k req/s
After: 118.3 req/s

+0.6% more req/s. Good work!

**Pull request merged**